source/agora/test/Base.d

```diff
@@ -139,9 +141,7 @@ public class TestNetwork : Network
             Thread.sleep(sleep_time);
         }
 
-        assert(fully_discovered.length == count,
-               format("Got %s/%s discovered nodes. Missing nodes: %s",
-                   fully_discovered.length, this.apis.length, this.todo_addresses));
```

Comment:
Honestly I don't see much of an improvement here... We do have stack traces so the assert being here shouldn't be a problem?

Reply:
Aren't stack traces still broken?

Reply:
```
core.exception.AssertError@source/agora/test/Base.d(145): Got 4/5 discovered nodes. Missing nodes: ["GD6GUK2NDHNTAS5TAKCIE6BFSLG5WELOE2GLRMZTORCY4FPWZOY5EPAZ":true, "GCI6VCQJG53N3HWQVENP6WED56YREJTDRQOW2VC2UBEADCST42GDE3QB":true, "GBDESH4FGFCEMTDADJMCFKPCHHZSFKZ4U5BYLX6NIB6ED3RDVZ23UE2Z":true, "GBCJQKB4QRTPUX4Y2H6PAKKRKIKH7RV4K5PK5CGTUI2EDH2SD3NAQY7Z":true]
----------------
??:? _d_assert_msg [0x1026c86c6]
/usr/local/opt/dmd/include/dlang/dmd/object.d:2911 void agora.test.Base.TestNetwork.waitUntilConnected(ulong) [0x1023c8051]
/usr/local/opt/dmd/include/dlang/dmd/object.d:2911 void agora.test.TestNetwork.__unittest_L19_C1() [0x1023ca805]
source/agora/app.d:36 void agora.test.TestNetwork.__modtest() [0x10234e5e8]
??:? int core.runtime.runModuleUnitTests().__foreachbody2(object.ModuleInfo*) [0x1026c93fb]
??:? int object.ModuleInfo.opApply(scope int delegate(object.ModuleInfo*)).__lambda2(immutable(object.ModuleInfo*)) [0x1026bd9ce]
```

Yeah they're still broken.

Reply:
They are, but moving this responsibility to the caller does not appeal to me too much... But okay, let's give it a try.